src/components/elements/input/RadioGroupInput.tsx

```diff
@@ -145,34 +146,40 @@ const RadioGroupInput = ({
           }}
           {...props}
         >
-          {options.map(({ code, label }) => (
-            <FormControlLabel
-              data-testid={`option-${code}`}
-              disabled={props.disabled}
-              value={code}
-              aria-label={label || code}
-              onClick={(e) => onClickOption(e, code)}
-              control={
-                <ControlComponent
-                  disabled={props.disabled}
-                  onKeyDown={onKeyDown}
-                  data-checked={value?.code === code ? true : false}
-                />
-              }
-              checked={value?.code === code ? true : false}
-              key={code}
-              label={label || code}
-              componentsProps={{
-                typography: {
-                  variant: 'body2',
-                  mr: 0.5,
-                  color:
-                    checkbox && value && value?.code !== code
-                      ? 'gray'
-                      : undefined,
-                },
-              }}
-            />
+          {options.map(({ code, label, helperText }) => (
+            <Fragment key={code}>
+              <FormControlLabel
+                data-testid={`option-${code}`}
+                disabled={props.disabled}
+                value={code}
+                aria-label={label || code}
+                onClick={(e) => onClickOption(e, code)}
+                control={
+                  <ControlComponent
```

Comment:
For accessibility this relies on being rendered inside the label. But we should probably link up the helper text to it too. I think something like`aria-labelledby="<main label element> <helper text element>" here?

Reply:
Good suggestion, I did a little work to improve accessibility of the input component; I also extracted the option out into its own component.
* removed rid of the enclosing MUI Form Group (it didn't seem to be needed)
* add fieldset and legend elements for screen readers
* add aria-labelledby for both label and helperText. (I suspect aria-describedby would be more accurate but I couldn't get that work with voiceover)